common/src/main/java/com/hedera/block/common/utils/FileUtilities.java

```diff
@@ -166,17 +171,38 @@ public static byte[] readGzipFileUnsafe(@NonNull final Path filePath) throws IOE
      * <p>
      * This method uses default extensions for gzip and block files.
      * <p>
-     * This method is _unsafe_ because it reads the entire file content into
-     * a single byte array, which can cause memory issues, and may fail if the
-     * file contains a large amount of data.
+     * This method is _unsafe_ because it reads the entire file content into a single byte array,
+     * which can cause memory issues, and may fail if the file contains a large amount of data.
+     *
+     * @param file to read bytes from
+     *
+     * @return byte array of the content of the file or null if the file extension is not supported
+     *
+     * @throws IOException      if unable to read the file.
+     * @throws OutOfMemoryError if a byte array large enough to contain the file contents cannot be
+     *                          allocated (either because it exceeds MAX_INT bytes or exceeds
+     *                          available heap memory).
+     */
+    public static byte[] readFileBytesUnsafe(@NonNull final File file) throws IOException {
+        return readFileBytesUnsafe(file.toPath());
+    }
+
+    /**
+     * Read a file and return the content as a byte array.
+     * <p>
+     * This method uses default extensions for gzip and block files.
+     * <p>
+     * This method is _unsafe_ because it reads the entire file content into a single byte array,
+     * which can cause memory issues, and may fail if the file contains a large amount of data.
      *
      * @param filePath Path to the file
-     * @return byte array of the content of the file or null if the file extension is not
-     *     supported
-     * @throws IOException if unable to read the file.
-     * @throws OutOfMemoryError if a byte array large enough to contain the
-     *     file contents cannot be allocated (either because it exceeds MAX_INT
-     *     bytes or exceeds available heap memory).
+     *
+     * @return byte array of the content of the file or null if the file extension is not supported
+     *
+     * @throws IOException      if unable to read the file.
+     * @throws OutOfMemoryError if a byte array large enough to contain the file contents cannot be
+     *                          allocated (either because it exceeds MAX_INT bytes or exceeds
+     *                          available heap memory).
      */
     public static byte[] readFileBytesUnsafe(@NonNull final Path filePath) throws IOException {
         return readFileBytesUnsafe(filePath, ".blk", ".gz");
```

Comment:
should this use the Constants instead of hardcoding the values `.blk` and `.gz`?

Reply:
@AlfredoG87 good question. I was thinking about that today when reviewing #300, since that issue should IMO use this logic  (see [this](https://github.com/hashgraph/hedera-block-node/pull/300#discussion_r1814782347) comment in the review there). That being said, we used to have both these strings as constants in the beginning, but after a conversation we had with @jsync-swirlds and @mattp-swirldslabs we decided to remove them, since `.blk` is not generic and `.gz` we did not see any point in having, which I agree with both.

 The reason you see that implemented like that is because this is a single use, does not require a constant and is intended to be here only and not visible to the outside world (@jsync-swirlds actually finished that part after I left a comment on that method, I believe this was the thought process, correct me if I am wrong). And this method was extracted from a previously existing one and it was implemented with that if-else check for the extensions. I agree that currently this is maybe not very generic because of the `.blk` extension check.
 
I would propose that this can be utilized now by @jasperpotts for [this](https://github.com/hashgraph/hedera-block-node/pull/300#discussion_r1814782347) comment. and I think that the method should be refactored to **not** check for `.blk` but to be in the same fashion of what @jasperpotts currently has at 6bbb15975b6821a09cc688c252d39866912646fd from my comment, and to also ommit the `null` check. Personally I would propose the following change to the `readFileByteUnsafe`:
 
```java
    public static byte[] readFileBytesUnsafe(
            @NonNull final Path filePath,)
            throws IOException {
        final String filePathAsString = Objects.requireNonNull(filePath).toString();
        Objects.requireNonNull(".gz");
        if (filePathAsString.endsWith(gzipFileExtension)) {
            return readGzipFileUnsafe(filePath);
        } else {
            return Files.readAllBytes(filePath);
        }
    }
```

and of course we remove the overloaded method that already exists since they have now the same parameters.

This seems more generic to me and better fits the `common` module. @jsync-swirlds @mattp-swirldslabs @AlfredoG87  @jasperpotts What do you guys think about that change? I think is better to be done and @jasperpotts  can use the generic logic in his `tools` now. Let me know

Reply:
None of these extensions is likely to remain in place forever.  The whole point here is to have a method that accepts parameters for file extensions and offer a sane default via an override.  The values are hardcoded in the override just to avoid breaking existing code.  Once existing code changes to call the "parameterized" version, this convenience method can be removed entirely.

Let's _not_ (ever) rewrite things as a single method that hardcodes things like `.gz` extensions.

Reply:
@jsync-swirlds I do share your concern for breaking the existing logic if we change the implementation, I did mention that in my comment, I do not believe that if we change to an `if-else` we will currently break anything (but need to check thoroughly) but yet it is indeed dangerous to change the implementation in such a drastic way. So yes, based on that I do tend to agree that this should remain implemented as is and not get changed. @AlfredoG87 hope that resolves it, from my side it does, the danger of breaking something (even if it might not be visible now) is enough for me to drop the proposal.

Reply:
@jsync-swirlds noted, the context that you give changes everything, and that context was missing for me, hence the misunderstanding on my side as well as for the implementation details in #272 :)

That being said, I agree with you and the read method should have by default the ability to provide as arguments the suffix.